R/R/data_interface.R

```diff
@@ -439,6 +439,52 @@ sedona_read_shapefile <- function(sc,
     new_spatial_rdd(NULL)
 }
 
+
+#' Read a geoparquet file into a Spark DataFrame.
+#' Read a geoparquet file into a Spark DataFrame. The created dataframe is automatically registered.
+#'
+#' @param sc A \code{spark_connection}.
+#' @param location Location of the data source.
+#' @param name The name to assign to the newly generated table.
+#'
+#'
+#' @return A SpatialRDD.
+#'
+#' @examples
+#' library(sparklyr)
+#' library(apache.sedona)
+#'
+#' sc <- spark_connect(master = "spark://HOST:PORT")
+#'
+#' if (!inherits(sc, "test_connection")) {
+#'   input_location <- "/dev/null" # replace it with the path to your input file
+#'   rdd <- sedona_read_geoparquet(sc, location = input_location)
+#' }
+#'
+#' @family Sedona data interface functions
+#'
+#' @export
+sedona_read_geoparquet <- function(sc,
```

Comment:
Can we add a function that takes any format string? This way, the R interface will be able to automatically invoke any new reader we add in the future?

The example usage could be:

Read GeoParquet

```
sdf = sedona_read(sc, "geoparquet", location, name)
```

Read GeoTiff

```
sdf = sedona_read(sc, "geotiff", location, name)
```

Reply:
Thanks for the explanation. Makes sense to me. Would you please add one more reader/writer `sedona_read_geotiff` and `sedona_write_geotiff`?

It works the same way as the geoparquet: https://sedona.apache.org/latest-snapshot/api/sql/Raster-loader/#geotiff-dataframe-loader.

If you think this is too complicated, we certainly can leave it to another PR.

Reply:
Regarding the Geotiff: I started working on it. Reading the geotiff works fine.
There is one complexity given the created dataframe is a nested object. There is a package to help with that case: `sparklyr.nested`. I'll need more time to figure out whether we need to list it as a dependency or just suggest its use. And if it is required for the testing.